pkg/controller/endpointslice/utils.go

```diff
@@ -158,8 +158,8 @@ func newEndpointSlice(service *corev1.Service, endpointMeta *endpointMeta) *disc
 	ownerRef := metav1.NewControllerRef(service, gvk)
 	return &discovery.EndpointSlice{
 		ObjectMeta: metav1.ObjectMeta{
-			Labels:          map[string]string{serviceNameLabel: service.Name},
-			GenerateName:    fmt.Sprintf("%s.", service.Name),
+			Labels:          map[string]string{discovery.LabelServiceName: service.Name},
+			GenerateName:    fmt.Sprintf("%s-", service.Name),
 			OwnerReferences: []metav1.OwnerReference{*ownerRef},
```

Comment:
this is still setting owner references... what makes that appropriate here and not elsewhere?

Reply:
Owner references will still be set by the EndpointSlice controller, they are still valuable for garbage collection here and in normal use cases. The big change in this PR involves not setting them for the special use case that is the apiserver managing resources referring to itself (outside of the scope of the EndpointSlice controller or Endpoints controller).